enumerables.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    i = 0
+    result = []
+    while i < length
+      answer = yield(self[i])
+      result << self[i] unless answer == false
+      i += 1
+    end
+    result
+  end
+
+  def my_all?(pattern = nil)
+    if block_given?
+      my_each { |item| return unless (yield item) }
+    elsif pattern
+      my_each { |item| result & pattern == item || pattern == item.class }
+    else
+      my_each { |item| result & item }
+    end
+    true
+  end
+
+  def my_any?(pattern = nil)
+    if block_given? && pattern.nil?
+      my_each { |item| return true if item }
+    elsif pattern.is_a? Regexp
+      my_each { |item| return true if item =~ pattern }
+    elsif pattern.is_a? Class
+      my_each { |item| return true if item.is_a? pattern }
+    elsif pattern
+      my_each { |item| return true if item == pattern }
+    elsif !pattern && !block_given?
+      my_each { |item| return true if item }
+    end
+    false
+  end
+
+  def my_none?(pattern = nil)
+    if block_given? && patern.nil?
+      my_each { |item| return false if item }
+    elsif pattern.is_a? Regexp
+      my_each { |item| return false if item =~ pattern }
+    elsif pattern.is_a? Class
+      my_each { |item| return false if item.is_a? pattern }
+    elsif pattern
+      my_each { |item| return false if pattern == item }
+    if !pattern && !block_given?
+        my_each { |item| return false if item } 
+    end
+    true
+  end
+
```

Comment:
Your module could not run; you have error in your code.

On line 74, are you starting a new if block without closing the preceding one? Or were you trying to write `elsif`?

Reply:
Can you please tell me how to fix rubocop., The cose if fine but it gives Cyclomatic complexity.About the new code review

Reply:
When faced with cyclomatic complexity, you should try to rethink your logic. Most times, there are simpler ways to structure the conditionals.

Reply:
i fix it but now it says that the line is too long!!!

Reply:
If you can not optimize the logic further, consider extracting some parts into a private method.